onivim/oni
# Tabs show modified state.

Some early work on getting tabs to show the modified state and fix #977.

Current issue is that I'm using the `buffer.modified` value to check if a buffer has been modified, which is what the `buffer` mode tabs use.

However, this seems to not be set when swapping between splits or between tabs. This also seems to be an issue for me if I use `"tabs.mode": "buffer"` as well though, so is this a regression or was it just not wired up like that in the first place?

To test this I'm doing as follows :
- Open Oni
- Open file
- Open new file with `:vsplit FILE2`
- Edit file 2
- Click on original file, and modified status will go away in the Oni visualised tabs. Neovim returns 1 as expected when using `:echo &modified`.

This happens in both modes, and on the latest release, so I don't think its me, but if someone could confirm its not any of my settings that would be useful.

Regardless, to do:

- [x] Check where I am calling `tabpagebuflist` makes sense.
- [x] Fix the modified status turning off when swapping. 
- [x] Add Unit tests for Tab modified status function
- [x] Add CI test for Tab modified (and Buffer if needed)
